protocol.md

```diff
@@ -726,6 +727,141 @@ Upload-Length: 11
+
+The value of `Upload-Challenge` is defined as follows:
+
+```
+Upload-Challenge = SHA256([Upload-Secret] + SHA256(`[HTTP Method]` + [Upload-Offset] + [Upload-Secret] + [Content-Length]))
+```
+
+The *effective* HTTP method of the request that the `Upload-Challenge` is sent with MUST be
+used in place of `HTTP Method`. If `X-HTTP-Method-Override` is used in the same request,
+`HTTP method` needs to be replaced with the value of `X-HTTP-Method-Override`.
+
+Where `Upload-Offset` and `Content-Length` headers are used in the same request, their
+values MUST be used as the respective `Upload-Offset` and `Content-Length` values. A
+value of `0` MUST be used for `Upload-Offset` and `Content-Length` where no headers
+of the same name are sent with the request.
+
+Servers receiving a `HEAD` or `PATCH` request with a missing or invalid `Upload-Challenge` value
+MUST respond with a `404 Not Found` status.
+
+For requests that reference multiple upload resources, the `Upload-Challenge` is computed by
+concatenating the `Upload-Challenge`s of the individual upload resources for the request in the
+order they are referenced. The SHA256 checksum computed over the concatenated `Upload-Challenge`s
+is then used as the `Upload-Challenge` for the request:
+
+```
+Upload-Challenge = SHA256([Upload-Challenge for resource 1] + … + [Upload-Challenge for resource n])
+```
+
+#### Examples
+
+An upload resource is created with [Creation With Upload](#creation-with-upload) and an `Upload-Secret`:
+
+```
+POST /files HTTP/1.1
+Upload-Length: 100
+Upload-Secret: R290Y2hhISBVc2UgYW4gYWN0dWFsIGhpZ2gtZW50cm9weSBzb3VyY2UgaW4gcmVhbCBjb2RlIDspCg
+Content-Length: 5
+Content-Type: application/offset+octet-stream
+Tus-Resumable: 1.0.0
+
+hello
+```
+
+The upload is continued with a `PATCH` request, including an `Upload-Challenge` computed as
+
+```
+Upload-Challenge = SHA256("R290Y2hhISBVc2UgYW4gYWN0dWFsIGhpZ2gtZW50cm9weSBzb3VyY2UgaW4gcmVhbCBjb2RlIDspCg" + SHA256("PATCH" + "5" + "R290Y2hhISBVc2UgYW4gYWN0dWFsIGhpZ2gtZW50cm9weSBzb3VyY2UgaW4gcmVhbCBjb2RlIDspCg" + "50"))
+```
+
+and included in the request:
+
+```
+PATCH /files/24e533e02ec3bc40c387f1a0e460e216 HTTP/1.1
+Upload-Offset: 5
+Upload-Challenge: ed9e63743b8073e97a1bd54d0f92161fa32c27bff130f11ff896e6e0271f2688
+Content-Length: 50
+Content-Type: application/offset+octet-stream
+Tus-Resumable: 1.0.0
+
+[50 bytes of content]
+```
+
+Then, a `DELETE` request for the upload is sent, including an `Upload-Challenge` computed as
+
+```
+Upload-Challenge = SHA256("R290Y2hhISBVc2UgYW4gYWN0dWFsIGhpZ2gtZW50cm9weSBzb3VyY2UgaW4gcmVhbCBjb2RlIDspCg" + SHA256("DELETE" + "0" + "R290Y2hhISBVc2UgYW4gYWN0dWFsIGhpZ2gtZW50cm9weSBzb3VyY2UgaW4gcmVhbCBjb2RlIDspCg" + "0"))
+```
+
+and included in the request:
+
+```
+DELETE /files/24e533e02ec3bc40c387f1a0e460e216 HTTP/1.1
+Content-Length: 0
+Upload-Challenge: 59ca949f7eed4c4a1f7e40da7c119fb7cb1095262a12eab770685711ea5a176a
+Tus-Resumable: 1.0.0
+
+[50 bytes of content]
```

Comment:
Copy paste error? DELETE usually does not contain any body.

Reply:
Definitely a copy paste error. Thanks! :)

I will address this with the next commit.

Reply:
I just pushed an update fixing this.